tmt/steps/provision/__init__.py

```diff
@@ -1103,6 +1109,59 @@ def requires(cls) -> list['tmt.base.Dependency']:
         """ All requirements of the guest implementation """
         return []
 
+    def acquire_log(self, log_name: str) -> Optional[str]:
+        """
+        Fetch and return content of a log.
+        :param log_name: name of the log.
+        :returns: content of the log.
+        """
+        raise NotImplementedError
+
+    def store_log(
+            self,
+            log_path: Path,
+            log_content: str,
+            log_name: Optional[str] = None) -> None:
+        """
+        Save log content to a file.
+        :param log_path: a path to save into,could be a directory
+        or a file path.
+        :param log_content: content of the log.
+        :param log_name: name of the log, if not set, log_path
+        is supposed to be '/dev/null' or  a file path.
```

Comment:
I think this special handling of `/dev/null` is a needless complication. It's a file like any other. E.g. the test below, `if not log_path.is_dir() or str(log_path) == '/dev/null':` - the check for `/dev/null` will never be performed, because `/dev/null` is not a directory, and that's enough for the condition to be true.

Let's ask a question, will there be a special branch for other magic files, like `/dev/sdb` or `/dev/zero`? :) I see no advantage, why do you think it's needed for `/dev/null`? If the user decides to not save guest logs, there would probably be an option that would tell tmt to not even call `handle_guest_logs()`. It would be much easier to not save logs at all rather than fetching them and dumping them into `/dev/null`. And as I said, it's a file like any other anyway...